### PR TITLE
Add pdf filling support for button types

### PR DESCRIPTION
**Pull request by gracetaii**

### CODE REVIEW

File: fillpdf.go [hunk 1]
```diff
@@ -172,12 +194,13 @@ func createFdfFile(form Form, path string) error {
 	// Write the form data.
 	var valueStr string
 	for key, value := range form {
-		// Convert to Latin-1.
-		valueStr, err = latin1Encoder.String(fmt.Sprintf("%v", value))
+		// First converts the FormOption value to the correct fdf string representation.
+		// Then converts string to Latin-1.
+		valueStr, err = latin1Encoder.String(fmt.Sprintf("%v", value.String()))
```
Comment (nkansal96) [inline fillpdf.go]: I think this can be simplified

```suggestion
		valueStr, err = latin1Encoder.String(value.String())
```

File: fillpdf.go [hunk 2]
```diff
@@ -36,9 +36,31 @@ var (
...
+// a form field accepts.
+type FormOption interface {
+	String() string
+}
+
+// TextOption is the FomrOption type for text fields.
+type TextOption string
+
+// String() returns the  fdf string representation of a text field. i.e. (textValue).
+func (t TextOption) String() string { return "(" + string(t) + ")" }
```
Comment (nkansal96) [inline fillpdf.go]: For readability
```suggestion
func (t TextOption) String() string {
  return "(" + string(t) + ")"
}
```

File: fillpdf.go [hunk 3]
```diff
@@ -36,9 +36,31 @@ var (
 	latin1Encoder = encoding.ISO8859_1.NewEncoder()
 )
 
+// FormOption is an interface representing the different types of values
+// a form field accepts.
+type FormOption interface {
+	String() string
+}
+
+// TextOption is the FomrOption type for text fields.
```
Comment (nkansal96) [inline fillpdf.go]:
```suggestion
// TextOption is the FormOption type for text fields.
```

File: fillpdf.go [hunk 4]
```diff
@@ -36,9 +36,31 @@ var (
...
+type TextOption string
+
+// String() returns the  fdf string representation of a text field. i.e. (textValue).
```
Comment (nkansal96) [inline fillpdf.go:48]:
```suggestion
// String() returns the FDF string representation of a text field. i.e. (textValue).
```

File: fillpdf.go [hunk 5]
```diff
@@ -36,9 +36,31 @@ var (
...
+func (t TextOption) String() string { return "(" + string(t) + ")" }
+
+// ButtonOption is the FormOption type for button fields.
+// The value of the button depends on the button type and the state options.
+// For example, the button may have states "On" and "Off" indicating whether or not the button is checked.
+type ButtonOption string
+
+// String() returns the  fdf string representation of a button. i.e. /buttonState.
```
Comment (nkansal96) [inline fillpdf.go:58]:
```suggestion
// String() returns the FDF string representation of a button. i.e. /buttonState.
```

**Review by nkansal96 — APPROVED** [review]

**Comment by nkansal96** [comment]
We should add a test for `createFDFFile`. We can make it easy to test by splitting up the generation of the FDF content from the actual writing to file.

**Comment by gracetaii** [comment]
> We should add a test for `createFDFFile`. We can make it easy to test by splitting up the generation of the FDF content from the actual writing to file.

will add it in a followup pr

**Pull request merged**